### ISSUE TITLE
Responsive layout

### ISSUE THREAD

**Issue by JosefJezek** [jitsi/jitsi-meet#335]
Please Responsive layout.

**Comment by emcho** [comment]
I don't understand this. Please send your recommendations or questions to out dev mailing list (https://jitsi.org/lists). and try to be more descriptive.

Thanks

**Comment by JosefJezek** [comment]
Responsive layout for web client https://meet.jit.si/ Support for mobile screens.

![image](https://cloud.githubusercontent.com/assets/2110319/9193321/bb34a126-4012-11e5-98bf-0598465eeb28.png)

![image](https://cloud.githubusercontent.com/assets/2110319/9193326/d1fb26dc-4012-11e5-8c23-43b734675dd1.png)

**Comment by marclaporte** [comment]
Done:

Jitsi Meet: Proposal to migrate to the Bootstrap front-end framework in a future version (responsive design, easy theming, future-proof)
http://lists.jitsi.org/pipermail/dev/2015-August/025018.html

Thanks!

**Comment by gpolitis** [comment]
I don't know how well this aligns with our roadmap but I would love to see this in meet. Maybe we should reopen this issue.

**Comment by marclaporte** [comment]
Bootstrap 4 alpha:
http://blog.getbootstrap.com/2015/08/19/bootstrap-4-alpha/

**Comment by marclaporte** [comment]
For the record, Bootstrap code is already included and used for:
    "tooltip": "./node_modules/bootstrap/js/tooltip.js",
    "popover": "./node_modules/bootstrap/js/popover.js",

https://github.com/jitsi/jitsi-meet/commit/f4da6784652fd546bd6261300635683985cb8da5

**Comment by akuckartz** [comment]
:+1:

**Comment by akuckartz** [comment]
Can this issue please be reopened? I tried to use meet.jit.si in Firefox on a Smartphone and it was not usable due to missing responsiveness. I can offer to proved feedback !

**Comment by marclaporte** [comment]
@akuckartz You are registered as a tester :-)

**Comment by marclaporte** [comment]
Here is an update on the mailing list:
http://lists.jitsi.org/pipermail/dev/2015-November/025588.html

Thanks!

**Comment by HelioGuilherme66** [comment]
Count with me as tester.

**Comment by Bugsbane** [comment]
I'd like to test too, thanks.

**Comment by luciash** [comment]
WIP: https://github.com/luciash/jitsi-meet-bootstrap

**Comment by marclaporte** [comment]
There was a Jitsi Meet community conference call on 2015-11-09. There is general interest and agreement that Bootstrap is the way to go. There is a concern on timing as some design improvements are planned. It could have been nice to do both at once, but it will be just as easy or likely easier to improve the design with Bootstrap, so there is no need to wait.

Next steps:
* Improve the state of the WIP implementation at https://github.com/luciash/jitsi-meet-bootstrap (luciash)
* Open up the demo server for community feedback (It's already running with auto-updates thanks to George)
* Improve implementation with feedback
* In parallel with above, figure out a solution for pulling in Bootstrap (ex.: Bower)
* When all is good, formalize the PR

Once the first version of Jisti Meet with Bootstrap is released, we'll be attentive to any regressions or issues it may cause, and deal with these.

Once all is nice and stable with vanilla Bootstrap, we'll look into adding Bootstrap Tour:
http://bootstraptour.com/

**Comment by Bugsbane** [comment]
So will the url for the demo server be posted on this issue? I'm keen to give feedback when it's available.

**Comment by marclaporte** [comment]
Yes, it will be posted here.

If anyone can help with coding and not just testing (which will need a bit later), just email me directly and I'll provide you with not-yet-public dev server.

Thanks!

**Comment by marclaporte** [comment]
Not forgotten ;-)  We got caught up on a project. Will post again here as soon as we have something testable. Thanks!

**Comment by emcho** [comment]
Thanks fornthe update Marc!

**Comment by luciash** [comment]
Yes, not forgotten! Thanks Marc for the update!

**Comment by marclaporte** [comment]
Hi all!

Sorry for the delay. Our big project was launched yesterday so we now get back to this.

Thanks!

**Comment by marclaporte** [comment]
Hi all!

Here is test site:
https://meet.evoludata.com:8443/testroom

The commits are here:
https://github.com/luciash/jitsi-meet-bootstrap/commits/master?author=luciash

Please test and report feedback here.

Please keep in mind that this is a not a major visual revamp. The goal at this step is to make usable on mobile devices (responsive, etc.) and move to Bootstrap. Later, we can proceed to to complete UI overhaul to allow the app to be "skinned" by Bootstrap themes. Also, configurable icon sets would be nice.

Thanks!

**Comment by marclaporte** [comment]
When screen is very narrow, horizontal top menu items go on a second line (which is fine) but background is missing.
![2016-01-31_092431](https://cloud.githubusercontent.com/assets/1004261/12702833/6d4dc164-c801-11e5-965a-fc35faedde1b.png)

**Comment by akuckartz** [comment]
@marclaporte Thanks for working on this! I confirm that issue in the screenshot.

**Comment by bgrozev** [comment]
@marclaporte This is great! Just a heads up in case you haven't seen: last week we switched to using lib-jitsi-meet as a dependency, so if you have substantial new changes coming, you may want to rebase your existing work to avoid tough merges later.

**Comment by luciash** [comment]
@marclaporte fixed in https://github.com/luciash/jitsi-meet-bootstrap/commit/908c3ec6b7b9b5f65eca69b0170958677a5f25a1
@bgrozev - Thanks for the comment! I am doing merges from upstream now and then so all should be fine I hope...

**Comment by marclaporte** [comment]
@gpolitis Thank you for the assistance with the demo server.
@luciash I confirm your fix works for me

**Comment by marclaporte** [comment]
Dialog box could be nicer:
![tempfileforshare_2016-02-07-18-33-19](https://cloud.githubusercontent.com/assets/1004261/12876191/df46bca8-cdc9-11e5-8401-67924f66250a.jpg)

**Comment by marclaporte** [comment]
Some text is chopped off:
![tempfileforshare_2016-02-07-18-43-29](https://cloud.githubusercontent.com/assets/1004261/12876223/132623d2-cdcb-11e5-9483-2c6aea45e6aa.jpg)

**Comment by marclaporte** [comment]
Current behavior is that film strip (with video of users) is pushed to the left when chat is opened. On test site now, the chat box overlaps. This could be intentional and acceptable in mobile mode by lack of space, but in desktop mode, the current behavior should be maintained.
![2016-02-07_185539](https://cloud.githubusercontent.com/assets/1004261/12876277/cf2a5c00-cdcc-11e5-8b36-582cbd52511c.png)

**Comment by marclaporte** [comment]
@JosefJezek @akuckartz @HelioGuilherme66 @Bugsbane @bgrozev @gpolitis @emcho 

Please test https://meet.evoludata.com:8443/testroom and report feedback here (for things not yet reported).

The commits are here: https://github.com/luciash/jitsi-meet-bootstrap/commits/master?author=luciash

It's not yet an ideal interface for mobile (especially when keyboard opens up for chat), but it has now become usable on mobile. I propose to give it 1 or 2 more weeks for feedback and tweaks, and then to merge in. In the mean time, if someone could resolve the Bootstrap dependency aspect (that we discussed on the initial conference call), that would be great!

Further enhancements should be done as part of a revamp of the design. Some ideas:
1- Have a hamburger (or equivalent) icon to access all actions
2- Make UI more uniform (why do some buttons open stuff in the middle and others on the right?)
3- Do not have two buttons to open / close chat interface
4- Make actions icons/buttons bigger on mobile

Thanks!

**Comment by luciash** [comment]
@marclaporte https://github.com/luciash/jitsi-meet-bootstrap/commit/6f3bc4fcc4d28ea6bd996b683c884322ab6db0e6 should fix the https://github.com/jitsi/jitsi-meet/issues/335#issuecomment-181140567

**Comment by marclaporte** [comment]
@luciash Will you be OK to deal with many users and avoid issues seen in this screenshot?
![2016-02-29-jitsi-meet-too-many-users](https://cloud.githubusercontent.com/assets/1004261/13610838/8e5665de-e52d-11e5-8e1a-3581f35091eb.png)

**Comment by luciash** [comment]
@marclaporte yes, something like that. maybe the dropped packets (connection quality info) icons could go there too? other idea would be to zoom-in (enlarge) the user thumbnail on mouse over/touch...

**Comment by luciash** [comment]
Hi,
in my fork in one of the recent commits (https://github.com/luciash/jitsi-meet-bootstrap/commit/64a5a0fc5c1dbbddce718b65f41144040f0cee8d) I have tried to refactor lot of the HTML in the index page to contain and conform with the Bootstrap layout classes to have more uniform and themeable UI. I temporarily introduced css/all-jitsi.css file (as I hate to inspect, debug, compile and try to find in which file the resulting minified CSS is stored, sorry :-p) and a indexb.html file where I made the changes (rather than messing up with the index.html so I can easily compare) and I wanted to test it on one of the free themes from the Bootswatch collection. I have chosen Cyborg theme (http://bootswatch.com/cyborg/) as it looks simmilar to current Jitsi Meet theme. Here is the result:
![jitsi meet 2016-03-16 00-11-56](https://cloud.githubusercontent.com/assets/4555037/13852563/f6761344-ec62-11e5-8641-cdd5b4d2b8a8.png)
You can test it live here: https://meet.evoludata.com:8443/indexb.html?testroom
Let me know what you think. The proper next steps would be probably add the Bootswatch themes via node_modules, add a variable to the config, so at the compiling time, one can choose/set path to their preferred Bootstrap theme and merge the changes to the main index.html file (and remove the temporary files).

Thanks,
luci

**Comment by marclaporte** [comment]
Some overlap here: 
![2016-03-21-jitsi-meet-overlap](https://cloud.githubusercontent.com/assets/1004261/13941466/4ba34590-efbe-11e5-8614-e51ac3ba94c9.png)

**Comment by marclaporte** [comment]
@luciash  Thank you for going beyond the basic Bootstrapification and proceeding like this.

@everyone: https://bootswatch.com/ is a popular source of FLOSS themes (MIT license). We could bundle a few or all 16 of them (This is what we do in Tiki Wiki CMS Groupware). At the very least, we should have at least one dark one and one light one. This is a great example of taking advantage of the Bootstrap ecosystem. There could be a theme picker as a user preference (and the same place where user set their name, and pick their microphone and camera)

Thanks!

**Comment by marclaporte** [comment]
Here is the pull request: https://github.com/jitsi/jitsi-meet/pull/585

**Comment by luciash** [comment]
knock knock! ;)

**Comment by singpolyma** [comment]
This is the one thing that keeps jitsi meet from replacing skype for me

**Comment by marclaporte** [comment]
@emcho: do you still want this?

**Comment by saghul** [comment]
I don't think it's a goal at this moment. @emcho please reopen if you think otherwise. We have spent considerable amount of time putting together mobile apps for Jitsi Meet, making the web UI mobile fiendly (past the welcome page, which already is) is not a goal anymore.
